internal/handlers/common/update_params.go

```diff
@@ -15,132 +15,63 @@
 package common
 
 import (
-	"fmt"
-
 	"go.uber.org/zap"
 
-	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
 	"github.com/FerretDB/FerretDB/internal/handlers/commonparams"
 	"github.com/FerretDB/FerretDB/internal/types"
-	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
-// UpdatesParams represents parameters for update command.
+// UpdatesParams represents parameters for the update command.
 type UpdatesParams struct {
-	DB         string
-	Collection string
-	Updates    []UpdateParams
+	DB         string         `ferretdb:"$db"`
+	Collection string         `ferretdb:"collection"`
+	Updates    []UpdateParams `ferretdb:"updates"`
+
+	Comment string `ferretdb:"comment,opt"`
+
+	Let *types.Document `ferretdb:"let,unimplemented"`
+
+	Ordered                  bool            `ferretdb:"ordered,ignored"`
+	BypassDocumentValidation bool            `ferretdb:"bypassDocumentValidation,ignored"`
+	WriteConcern             *types.Document `ferretdb:"writeConcern,ignored"`
 }
 
 // UpdateParams represents a single update operation parameters.
 type UpdateParams struct {
-	Filter  *types.Document
-	Update  *types.Document
-	Comment string
-	Multi   bool
-	Upsert  bool
+	// TODO: https://github.com/FerretDB/FerretDB/issues/2627
+	// get comment from query, e.g. db.collection.UpdateOne({"_id":"string", "$comment: "test"},{$set:{"v":"foo""}})
+	Filter *types.Document `ferretdb:"q,opt"`
+	Update *types.Document `ferretdb:"u,opt"`
+	Multi  bool            `ferretdb:"multi,opt"`
+	Upsert bool            `ferretdb:"upsert,opt"`
```

Comment:
Isn't his one `numericAsBool`?

Reply:
I don't see it's tested for those values. In `findAndModify` it's just bool.

Reply:
It's not tested but turns out it is `numericBool`.

```
db.testdata.runCommand({"findAndModify":"testdata","update":{"$set":{}}, "upsert":-50})
{
  lastErrorObject: {
    n: 1,
    updatedExisting: false,
    upserted: ObjectId("6461df27425e9da653295bf9")
  },
  value: null,
  ok: 1
}
```